goGene/goGene.go

```diff
@@ -0,0 +1,176 @@
+package goGene
+
+import (
+	"errors"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/vcf"
+)
+
+type Feature int32
+type MutationType byte
+
+const (
+	Intron   Feature = -1
+	UtrThree Feature = -3
+	UtrFive  Feature = -5
+	// All positive values refer to cDNA position
+
+	Silent       MutationType = 0
+	Missense     MutationType = 1
+	Nonsense     MutationType = 2
+	Frameshift   MutationType = 3
+	Intergenic   MutationType = 4
+	Intronic     MutationType = 5
+	Splice       MutationType = 6 // +/- 1-2 of E-I boundary
+	FarSplice    MutationType = 7 // +/- 3-10 of E-I boundary
+	DisruptStart MutationType = 8
+	DisruptStop  MutationType = 9
+)
+
+// GoGene is a processed version of a gtf record that enables easy
+// traversal and manipulation of genes on the genomic and mRNA levels.
+type GoGene struct {
+	id           string
+	startPos     int
+	strand       bool
+	cdsStarts    []int
+	cdsEnds      []int
+	genomeSeq    []dna.Base
+	cdnaSeq      []dna.Base
+	featureArray []Feature
+	orig         goGeneBackup
+	changeLog    []diff
+}
+
+type goGeneBackup struct {
+	startPos     int
+	cdsStarts    []int
+	cdsEnds      []int
+	genomeSeq    []dna.Base
+	cdnaSeq      []dna.Base
+	featureArray []Feature
+}
+
+type diff struct {
+	genomePos int
+	removed   []dna.Base
+	added     []dna.Base
+}
+
+type EffectPrediction struct {
+	Consequence MutationType
+	CdnaPos     int // zero base
+	CdnaOffset  int
+	AaPos       int // zero base
+	AaRef       []dna.AminoAcid
+	AaAlt       []dna.AminoAcid
+}
+
+// GenomicToCdna converts genomic coordinates to cDNA coordinates. The return format is c.100+10.
+// The first int return is the nearest position in the coding sequence.
+// The second int return is the distance from the nearest coding exon
+// (>0 if after cds; <0 if before cds, ==0 if inside coding sequence).
```

Comment:
Are "before" and "after" in terms of genomic coordinates, or in terms of direction of transcription?  For example, if the gene is on the minus strand, would an intronic position with a greater position number than its closest exon be before or after that exon?  Might also be good to mention what happens with ties.